build_resolvers/lib/src/resolver.dart

```diff
@@ -25,16 +25,24 @@ final _logger = Logger('build_resolvers');
 /// Implements [Resolver.libraries] and [Resolver.findLibraryByName] by crawling
 /// down from entrypoints.
 class PerActionResolver implements ReleasableResolver {
-  final ReleasableResolver _delegate;
-  final Iterable<AssetId> _entryPoints;
+  final AnalyzerResolver _delegate;
+  final BuildStep _step;
 
-  PerActionResolver(this._delegate, this._entryPoints);
+  Set<AssetId> _entryPoints;
+
+  PerActionResolver(this._delegate, this._step, Iterable<AssetId> entryPoints) {
+    _entryPoints = entryPoints.toSet();
+  }
 
   @override
   Stream<LibraryElement> get libraries async* {
     final seen = Set<LibraryElement>();
     final toVisit = Queue<LibraryElement>();
-    for (final entryPoint in _entryPoints) {
+
+    // keep a copy of entry points in case [_resolveIfNecessary] is called
+    // before this stream is done.
+    final entryPoints = Set.of(_entryPoints);
```

Comment:
nit: we can just create a list at this point which would be a bit cheaper (we already know there are no duplicates)

You can just use `.toList()` to get that.